### DISCUSSION TITLE
What actually I'm working in keyhunt

### DISCUSSION THREAD

**Discussion by albertobsd** [albertobsd/keyhunt#142] [Announcements]
Well, since I retake the keyhunt development there are a lot of request from the users, i can't attend all of it, but I'm working in some of those request.

**I am always going to solve critical bugs**

- Missing keys
- Buffer overflows or other vulnerabilities
- Other bugs that affect the performance of the program.

### What features I'm working on?

- Middleout mode for BSGS. See : https://github.com/albertobsd/keyhunt/issues/120
- Check bloom filter and bPtable  from disk see:  https://github.com/albertobsd/keyhunt/issues/129
- New Sequential search with some statistics to search only random look a like values See: https://github.com/albertobsd/keyhunt/issues/115
- Modes backward, dance and middleout for all the search: address, rmd160 and xpoint
- Option -S for all the modes: address, rmd160 and xpoint
- Pollard Rho
- Pure GMP version (again) to support all ARM hardware
- Save range already search in sequential and backward mode. See https://github.com/albertobsd/keyhunt/issues/144 

That list is not in sorted, I'm working first in the easiest ones of course because i'm some lazy xD

### Done
- ~~Add a third bloom filter check to use less memory in RAM.~~  (Finished) See : https://github.com/albertobsd/keyhunt/issues/139
- ~~New Minikeys sequential generator~~ (Finished) see: https://github.com/albertobsd/keyhunt/issues/140

If you want an extra feature that is not on the list please feel free to make a donation

https://github.com/albertobsd/keyhunt#donations

### Why I have not write my program for GPU yet?

Because I haven't one, I hope to have desktop computer with some decent GPU to start write my program also for GPU.

**Comment by albertobsd** [comment]
In the commit https://github.com/albertobsd/keyhunt/commit/338184301c155ba2d07bb6259f05f88543ca0a53 I added the 3rd bloom filter, this Bloom filter save 20% of RAM
Also that commit have between 2 to 3 times more speed for minikeys.

**Comment by ghost** [comment]
Hi @albertobsd,

I think the dance mode (-B dance) isn’t working properly in rmd160 search mode.
Even though the flag is accepted, the scanning behavior looks fully sequential, not mixed or bidirectional as expected.

Could you please confirm if dance mode is actually implemented for rmd160, or if it defaults to sequential stepping internally?

Thanks for your great work on KeyHunt!

Mahi

**Comment by 627xiaobai** [comment]
Do you have an ARM build?
